services/wallet/service.go

```diff
@@ -433,31 +433,30 @@ func (service *Service) LinkWallet(
 	// verify that the user is kyc from uphold. (for all wallet provider cases)
 	if uID, ok, c, err := wallet.IsUserKYC(ctx, transactionInfo.Destination); err != nil {
 		// get the rewards wallet id from the uphold wallet info
-		infoID, err := uuid.FromString(info.ID)
-		if err != nil {
-			return fmt.Errorf("failed to parse wallet id: %w", err)
+		infoID, infoIDErr := uuid.FromString(info.ID)
+		if infoIDErr != nil {
+			return fmt.Errorf("failed to parse uphold id: %w", err)
 		}
 		// check if this gemini accountID has already been linked to this wallet,
 		if errors.Is(err, errorutils.ErrInvalidCountry) {
-			ok, err := service.Datastore.HasPriorLinking(
+			ok, priorErr := service.Datastore.HasPriorLinking(
 				ctx, infoID, uuid.NewV5(ClaimNamespace, userID))
-			if err != nil {
-				return fmt.Errorf("failed to check prior linkings: %w", err)
+			if priorErr != nil {
+				return fmt.Errorf("failed to check prior linkings: %w", priorErr)
 			}
 			if !ok {
 				// then pass back the original geo error
 				return err
 			}
 			// allow invalid country if there was a prior linking
 		} else {
-			// if so ignore the region not supported error
-			return err
+			// not a invalid country error
+			// there was an unexpected error
+			return handlers.WrapError(err,
```

Comment:
Does returning the wrapped errors affect the controller call i.e. will we be wrapping twice

Reply:
addressed: https://github.com/brave-intl/bat-go/pull/1671/commits/9fd3136c1f3d30e39741617bbb05110368f56a6a